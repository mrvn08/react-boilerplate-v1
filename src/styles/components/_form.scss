.form{
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $regular;
    }
}

.form__error{
    margin: 0 0 $medium 0;
    font-style: italic;
}