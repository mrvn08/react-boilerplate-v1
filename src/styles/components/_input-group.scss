.input-group{
    display: flex;
    margin-bottom: $medium;

    @media (max-width: $mobile-breakpoint) {
        flex-direction: column;
        margin-bottom: $regular;
    }
}

.input-group__item{
    margin-right: $small;
    @media (max-width: $mobile-breakpoint) {
        margin: 0 0 $small 0;
        
        .text-input, .select, .textarea{
            width: 100%;
        }
    }
}

