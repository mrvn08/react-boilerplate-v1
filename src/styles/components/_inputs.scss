.text-input{
    border: 1px solid #cacccd;
    height: 50px;
    padding: $small;
    font-size: $font-size-large;
    font-weight: 300;
}

.select{
    @extend .text-input;
}

.textarea{
    @extend .text-input;
    height: 10rem;
}