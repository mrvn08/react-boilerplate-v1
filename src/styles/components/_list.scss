.list-header{
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $small $regular;
}

.list-body{
    margin-bottom: $medium;
    
    @media (max-width: $mobile-breakpoint) {
        margin-bottom: $regular;
    }
}

.list-item{
    align-items: center;
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    justify-content: space-between;
    padding: $regular;
    text-decoration: none;
    transition: background 0.3s ease-in;
    &:hover {
        background: $off-white;
    }

    @media (max-width: $mobile-breakpoint) {
        align-items: flex-start;
        flex-direction: column;
        padding: $small;
    }
}

.list-item--message{
    align-items: center;
    color: $grey;
    justify-content: center;
    padding: $regular;
    &:hover{
        background: none;
    }
}

.list-item__title{
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title{
    color: $grey;
    font-size: $font-size-small;
}

.list-item__data{
    margin: 0 0 0 $small;

    @media (max-width: $mobile-breakpoint) {
        margin: $small 0 0 0;
    }
}