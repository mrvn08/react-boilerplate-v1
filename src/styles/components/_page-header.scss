.page-header{
    background: $off-white;
    margin-bottom: $medium;
    padding: $medium 0;
}

.page-header__actions{
    margin-top: $regular;
}

.page-header__title{
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
}