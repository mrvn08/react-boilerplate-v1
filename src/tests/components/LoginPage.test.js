import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let startLogin, wrapper;
beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin}/>);
});

test('Should render login page correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toBeCalled();
})