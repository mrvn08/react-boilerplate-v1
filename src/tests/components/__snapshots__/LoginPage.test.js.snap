// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      React Boilerplate App
    </h1>
    <p>
      App tagline here
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
